libnmstate/nm/route.py

```diff
@@ -0,0 +1,147 @@
+#
+# Copyright 2019 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+import logging
+from operator import itemgetter
+from libnmstate.schema import LinuxRoute
+from libnmstate.schema import LinuxRouteOrigin
+
+from libnmstate import iplib
+
+
+def get_routes(client, skip_networks, func, auto_origin_value):
+    ret = []
+    for active_connection in client.get_active_connections():
+        ip_cfg = func['get_ip_cfg_func'](active_connection)
+        if not ip_cfg:
+            continue
+
+        iface_name = _get_iface_name(active_connection)
+        if not iface_name:
+            logging.warning(
+                'Got connection {} has not interface name'.format(
```

Comment:
Active connection {} has no interface name

Is this an error or a supported option? In what cases this may happen?

Reply:
No idea, I never got this. The code is ensure we could get a warning when strange things happens.

Reply:
I guess the question is: Why not explode?
This is especially true if we do not know, it makes it an unexpected case and we need to detect it somehow.
At the minimum I would log the exception itself (logging.exception), then consider if to ignore the error or make a noise. I tend here to explode and let the caller/user/client try again and open a bug.